src/vcpkg/commands.update-baseline.cpp

```diff
@@ -0,0 +1,146 @@
+        }
+        else if (r.kind == "builtin")
+        {
+            return get_baseline_from_git_repo(paths, builtin_git_url);
+        }
+        else
+        {
+            return nullopt;
+        }
+    }
+
+    void UpdateBaselineCommand::perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths) const
+    {
+        args.parse_arguments(COMMAND_STRUCTURE);
+
+        bool has_builtin_baseline = false;
+        bool config_in_manifest = false;
+        vcpkg::Configuration configuration = paths.get_configuration();
+
+        vcpkg::Path configuration_file;
+
+        auto opt_config_path = paths.get_configuration_path();
+        if (auto p_config_path = opt_config_path.get())
+        {
+            configuration_file = std::move(*p_config_path);
+        }
+
+        Json::Object manifest;
+        vcpkg::Path manifest_file;
+        auto opt_manifest = paths.get_manifest();
+        if (auto p_manifest = opt_manifest.get())
+        {
+            manifest = *p_manifest;
+            // paths.get_manifest().has_value() == paths.get_manifest_path().has_value()
+            manifest_file = paths.get_manifest_path().value_or_exit(VCPKG_LINE_INFO);
+            if (p_manifest->contains("builtin-baseline"))
+            {
+                has_builtin_baseline = true;
+            }
+            if (p_manifest->contains("vcpkg-configuration"))
+            {
+                // it should be impossible to have both a manifest vcpkg-configuration, and a regular
+                // vcpkg-configuration
+                Checks::check_exit(VCPKG_LINE_INFO, configuration_file.empty());
+                config_in_manifest = true;
+            }
+        }
+
+        if (configuration_file.empty() && !has_builtin_baseline && !config_in_manifest)
+        {
+            msg::println(Color::warning, msgUpdateBaselineNoConfiguration);
+            Checks::exit_success(VCPKG_LINE_INFO);
+        }
+
+        if (has_builtin_baseline)
+        {
+            auto new_baseline = get_baseline_from_git_repo(paths, builtin_git_url);
+            if (auto p = new_baseline.get())
+            {
+                manifest.insert_or_replace("builtin-baseline", std::move(*p));
+            }
+        }
+        else if (auto default_reg = configuration.default_reg.get())
+        {
+            auto new_baseline = get_latest_baseline(paths, *default_reg);
+            if (auto p = new_baseline.get())
+            {
+                default_reg->baseline = std::move(*p);
+            }
+        }
+
+        for (auto& reg : configuration.registries)
+        {
+            auto new_baseline = get_latest_baseline(paths, reg);
+            if (auto p = new_baseline.get())
+            {
+                reg.baseline = std::move(*p);
+            }
+        }
+
+        if (config_in_manifest)
+        {
+            manifest.insert_or_replace("vcpkg-configuration", configuration.serialize());
+        }
+        else
```

Comment:
```suggestion
        else if (!configuration_file.empty())
```

In general, I think all these conditionals need a re-work. I suggest checking whether a configuration file exists first and then try the fallbacks (embedded config, `builtin-baseline`).